internal/server/GameLoop.go

```diff
@@ -69,19 +72,50 @@ func (s *Server) LootboxCheckAndDistributions() {
 				agents := megabike.GetAgents()
 				totAgents := len(agents)
 
+				// Compute resource allocation share.
+				validVotes := totAgents
+				accVotes := make(map[uuid.UUID]float64)
 				for _, agent := range agents {
-					// this function allows the agent to decide on its allocation parameters
-					// these are the parameters that we want to be considered while carrying out
-					// the elected protocol for resource allocation
-					agent.SetAllocationParameters()
-
-					// in the MVP  the allocation parameters are ignored and
-					// the utility share will simply be 1 / the number of agents on the bike
-					utilityShare := 1.0 / float64(totAgents)
-					lootShare := utilityShare * lootbox.GetTotalResources()
-					// Allocate loot based on the calculated utility share
-					fmt.Printf("Agent %s allocated %f loot \n", agent.GetID(), lootShare)
-					agent.UpdateEnergyLevel(lootShare)
+					agentVote := agent.GetResourceVote()
+
+					// Check if vote is valid.
+					sum := 0.0
+					for _, vote := range agentVote {
+						sum += vote
+					}
+					if math.Abs(sum-1.0) > 1e-9 {
+						validVotes -= 1
+						fmt.Printf("Agent %s provided an invalid resourceVote\n. It's vote will be discarded.", agent.GetID())
+						continue
+					}
```

Comment:
Referenced in other PR: have the GetVotes() function throw an error if the sum is not 1, and skip iteration in this case